File: src/main/java/org/cobbzilla/util/jdbc/UncheckedSqlException.java
package org.cobbzilla.util.jdbc;

import java.sql.SQLException;

public class UncheckedSqlException extends RuntimeException {

    private final SQLException sqlException;

    @java.beans.ConstructorProperties({"sqlException"})
    public UncheckedSqlException(SQLException sqlException) {
        this.sqlException = sqlException;
    }

    @Override public String getMessage() { return sqlException.getMessage(); }

    @Override public String getLocalizedMessage() { return sqlException.getLocalizedMessage(); }

    @Override public synchronized Throwable getCause() { return sqlException.getCause(); }

    @Override public synchronized Throwable initCause(Throwable throwable) { return sqlException.initCause(throwable); }

    @Override public String toString() { return sqlException.toString(); }

}

File: src/main/java/org/cobbzilla/util/jdbc/UncheckedSqlException.kt
package org.cobbzilla.util.jdbc

import java.sql.SQLException

class UncheckedSqlException @java.beans.ConstructorProperties("sqlException")
constructor(private val sqlException: SQLException) : RuntimeException() {

    override fun getMessage(): String {
        return sqlException.message
    }

    override fun getLocalizedMessage(): String {
        return sqlException.localizedMessage
    }

    @Synchronized
    override fun getCause(): Throwable {
        return sqlException.cause
    }

    @Synchronized
    override fun initCause(throwable: Throwable): Throwable {
        return sqlException.initCause(throwable)
    }

    override fun toString(): String {
        return sqlException.toString()
    }

}
